nodeadm/api/v1alpha1/nodeconfig_types.go

```diff
@@ -68,6 +68,12 @@ type KubeletOptions struct {
 	// Flags are [command-line `kubelet` arguments](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/).
 	// that will be appended to the defaults.
 	Flags []string `json:"flags,omitempty"`
+
+	// ImageCredentialProviderConfig is a custom Go template to configure
+	// the image credential provider. If not set, a default template is used.
+	// The template is rendered with the ConfigApiVersion, EcrProviderName
+	// and ProviderApiVersion variables.
+	ImageCredentialProviderConfig string `json:"imageCredentialProviderConfig,omitempty"`
```

Comment:
I think we'll want to use stronger typing than a go template, could we import `CredentialProvider`: https://kubernetes.io/docs/reference/config-api/kubelet-config.v1/#kubelet-config-k8s-io-v1-CredentialProvider

Or define an equivalent type if we need to handle multiple `apiVersion`-s of `CredentialProvider`?